File: keep_alive.py
import os
import datetime
from flask import Flask
from threading import Thread

app = Flask('')

@app.route('/')
def home():
    print(f"[{datetime.now().strftime('%Y-%m-%d %H:%M:%S')}] Ping received from {request.remote_addr}")
    print(f"staying alive")
    return "I'm alive!"

def run():
    port = int(os.environ.get("PORT", 8080))  # Render assigns a dynamic port
    app.run(host='0.0.0.0', port=port)

def keep_alive():
    t = Thread(target=run)
    t.start()

File: keep_alive.rb
# frozen_string_literal: true

require 'sinatra/base'

# Simple Sinatra app to keep the bot alive on platforms like Render.com
class KeepAlive < Sinatra::Base
  set :port, ENV['PORT'] || 4567
  set :bind, '0.0.0.0'

  get '/' do
    'Bot is alive! ✅'
  end
end

# Start the server in a separate thread
Thread.new do
  KeepAlive.run!
end

puts "🌐 KeepAlive server started on port #{ENV['PORT'] || 4567}"
